test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1

```diff
@@ -0,0 +1,1371 @@
+        "WindowsCurrentVersion",
+        "WindowsEditionId",
+        "WindowsInstallationType",
+        "WindowsProductId",
+        "WindowsProductName",
+        "WindowsRegisteredOrganization",
+        "WindowsRegisteredOwner",
+        "WindowsSystemRoot",
+        "WindowsVersion")
+   
+    if ([System.Management.Automation.Platform]::IsIoT)
+    {
+        Write-Verbose -Verbose -Message "WindowsInstallDateFromRegistry is not supported on IoT."
+    }
+    else
+    {
+        $WindowsPropertyArray += "WindowsInstallDateFromRegistry"
+    }
+
+    $propertyNames += $WindowsPropertyArray
+    
+    return $propertyNames
+}
+
+function New-ExpectedComputerInfo
+{
+    param([string[]]$propertyNames)
+
+    ## P-INVOKE TYPE DEF START ******************************************
+    function Get-FirmwareType
+    {
+$signature = @"
+[DllImport("kernel32.dll")]
+public static extern bool GetFirmwareType(ref uint firmwareType);
+"@
+        Add-Type -MemberDefinition $signature -Name "Win32BiosFirmwareType" -Namespace Win32Functions -PassThru
+    }
+
+    function Get-PhysicallyInstalledSystemMemory
+    {
+$signature = @"
+[DllImport("kernel32.dll")]
+[return: MarshalAs(UnmanagedType.Bool)]
+public static extern bool GetPhysicallyInstalledSystemMemory(out ulong MemoryInKilobytes);
+"@
+        Add-Type -MemberDefinition $signature -Name "Win32PhyicallyInstalledMemory" -Namespace Win32Functions -PassThru
+    }
+
+
+    function Get-PhysicallyInstalledSystemMemoryCore
+    {
+$signature = @"
+[DllImport("api-ms-win-core-sysinfo-l1-2-1.dll")]
+[return: MarshalAs(UnmanagedType.Bool)]
+public static extern bool GetPhysicallyInstalledSystemMemory(out ulong MemoryInKilobytes);
+"@
+        Add-Type -MemberDefinition $signature -Name "Win32PhyicallyInstalledMemory" -Namespace Win32Functions -PassThru
+    }            
+
+    function Get-PowerDeterminePlatformRole
+    {
+$signature = @"
+[DllImport("Powrprof", EntryPoint = "PowerDeterminePlatformRoleEx", CharSet = CharSet.Ansi)]
+public static extern uint PowerDeterminePlatformRoleEx(uint version);
+"@
+        Add-Type -MemberDefinition $signature -Name "Win32PowerDeterminePlatformRole" -Namespace Win32Functions -PassThru
+    }
+
+    function Get-LCIDToLocaleName
+    {
+$signature = @"
+[DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+public static extern int LCIDToLocaleName(uint localeID, System.Text.StringBuilder localeName, int localeNameSize, int flags); 
+"@
+        Add-Type -MemberDefinition $signature -Name "Win32LCIDToLocaleNameDllName" -Namespace Win32Functions -PassThru
+    }
+
+
+
+        ## P-INVOKE TYPE DEF END ******************************************
+
+        ## HELPER METHODS RELYING ON P-INVOKE BEGIN ******************************************
+    function Get-BiosFirmwareType
+    {
+        [int]$firmwareType = 0
+        (Get-FirmwareType)::GetFirmwareType([ref]$firmwareType)
+        return $firmwareType
+    }
+
+       
```

Comment:
There are inconsistencies in number of new lines between functions. Some use 1 line space, some use 2, etc.